graphics/blit/blit-scale.cpp

```diff
@@ -74,20 +75,24 @@ static void scaleNN(byte *dst, const byte *src,
 	const int dstIncY = (flipy ? -static_cast<int>(dstPitch) : static_cast<int>(dstPitch));
 
 	if (flipx) {
-		dst += (dstW - 1) * sizeof(Size);
+		dst += (dstW - 1) * Size;
 	}
 
 	if (flipy) {
 		dst += (dstH - 1) * dstPitch;
 	}
 
 	for (uint32 y = 0, yoff = 0; y < dstH; y++, yoff += srcIncY) {
-		const Size *srcP = (const Size *)(src + ((yoff >> 16) * srcPitch));
-		Size *dst1 = (Size *)dst;
+		const byte *srcP = src + ((yoff >> 16) * srcPitch);
+		byte *dst1 = dst;
 		for (uint32 x = 0, xoff = 0; x < dstW; x++, xoff += srcIncX) {
-			Size val = srcP[xoff >> 16];
-			*dst1 = val;
-			dst1 += dstIncX;
+			const byte *src1 = srcP + ((xoff >> 16) * Size);
+			if (Size == sizeof(Color)) {
+				*(Color *)dst1 = *(const Color *)src1;
```

Comment:
Is this endian safe?

Reply:
It doesn't matter in this case, we copy 4 bytes as they are. Endianness is retained.

Reply:
Indeed, I have to stop reading code when I'm tired...